### ISSUE TITLE
(cli): cdk migrate fails to process Fn::Sub within Fn::ImportValue

### ISSUE THREAD

**Issue by danielfen** [aws/aws-cdk#29014]
### Describe the bug

It is common to use the intrinsic function Fn::Sub within Fn::ImportValue, to import cross-stack values in an environment-specific way (e.g. selecting on a dev, test or production environment stack export). An example of this syntax is given on the `Fn::ImportValue` dcumentation: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-importvalue.html#w9aac34c28c46c15

The CDK Migrate CLI command fails currently when processing templates containing this syntax with the error

```
Migrate failed for `<stack name>`: stack generation failed due to error '<Resource Path>.Fn::ImportValue: invalid type: map, expected a string at ...'

stack generation failed due to error '<Resource Path>.Fn::ImportValue: invalid type: map, expected a string at ...'
```

### Expected Behavior

cdk migrate to be able to generate CDK code for templates containing

```
"Fn::ImportValue": {
  "Fn::Sub": [
    "${Value}...",
    {
      "Value": {
        "Ref": "ParameterName"
      }
    }
  ]
}
```

### Current Behavior

An error raised due to the Fn::Sub within Fn::ImportValue not being a string value

### Reproduction Steps

Run `cdk migrate --stack-name my-stack --language typescript --from-path <template> --output-path <path for output>` using the following template:
```
{
  "AWSTemplateFormatVersion": "2010-09-09",
  "Parameters": {
    "EnvironmentName": {
      "Description": "Environment used for this deployment.",
      "Type": "String",
      "AllowedValues": [
        "dev",
        "stage",
        "prod"
      ]
    }
  },
  "Resources": {
    "MyApi5xxErrorsAlarm": {
      "Type": "AWS::CloudWatch::Alarm",
      "Properties": {
        "AlarmDescription": "Example alarm",
        "Namespace": "AWS/ApiGateway",
        "Dimensions": [
          {
            "Name": "ApiName",
            "Value": "MyApi"
          }
        ],
        "MetricName": "5XXError",
        "ComparisonOperator": "GreaterThanThreshold",
        "Statistic": "Average",
        "Threshold": 0,
        "Period": 900,
        "EvaluationPeriods": 1,
        "TreatMissingData": "notBreaching",
        "AlarmActions": [
          {
            "Fn::ImportValue": {
              "Fn::Sub": [
                "${Environment}AlarmsTopicArn",
                {
                  "Environment": {
                    "Ref": "EnvironmentName"
                  }
                }
              ]
            }
          }
        ]
      }
    }
  }
}
```
The error will be raised

Run the same command with the following template and the template is processed successfully:

```
{
  "AWSTemplateFormatVersion": "2010-09-09",
  "Parameters": {
    "EnvironmentName": {
      "Description": "Environment used for this deployment.",
      "Type": "String",
      "AllowedValues": [
        "dev",
        "stage",
        "prod"
      ]
    }
  },
  "Resources": {
    "MyApi5xxErrorsAlarm": {
      "Type": "AWS::CloudWatch::Alarm",
      "Properties": {
        "AlarmDescription": "Example alarm",
        "Namespace": "AWS/ApiGateway",
        "Dimensions": [
          {
            "Name": "ApiName",
            "Value": "MyApi"
          }
        ],
        "MetricName": "5XXError",
        "ComparisonOperator": "GreaterThanThreshold",
        "Statistic": "Average",
        "Threshold": 0,
        "Period": 900,
        "EvaluationPeriods": 1,
        "TreatMissingData": "notBreaching",
        "AlarmActions": [
          {
            "Fn::ImportValue": "devAlarmsTopicArn"
          }
        ]
      }
    }
  }
}
```

### CDK CLI Version

2.126.0 (build fb74c41)

### Node.js Version

v18.17.1

### OS

20.04.6 LTS (Focal Fossa)

### Language

TypeScript

**Comment by tim-finnigan** [comment]
Thanks for reporting this issue, I could reproduce the error.

**Comment by TheRealAmazonKendra** [comment]
Looks like the recursion doing the translating is not going deep enough. Adding this task to cdk-from-cfn for fixing.

**Comment by TheRealAmazonKendra** [comment]
The fix for this was released in v2.132.
